force-app/main/default/classes/SOQL.cls

```diff
@@ -313,19 +428,28 @@ public with sharing class SOQL {
      * @param List of cloned Opportunity Ids
      */
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
-        // Query the original Opportunity Line Items
-
-        // Create a list to hold the new line items
-
-        // Generate a random number between 1 and 3
-
-        // Loop through each cloned Opportunity Id
-
-            // Loop through each original line item
-
-                // Clone the original line item and add the random number to the quantity
-
-        // Insert the new line items
+        List<OpportunityLineItem> oppLineItems= [
+            SELECT Id, OpportunityId, Quantity, PricebookEntryId, UnitPrice
+            FROM OpportunityLineItem
+            WHERE OpportunityId = :origOppId
+            ];
+
+        List<OpportunityLineItem> newLineItems = new List<OpportunityLineItem>();
+
+        Integer genNum = Integer.valueOf(Math.random() * 3 + 1);
```

Comment:
This probably should be within in the for loop so it is a random quantity every time. The instructions are a little unclear here and ultimately doesnt really matter.